apps/api/src/app/integrations/usecases/remove-integration/remove-integration.usecase.ts

```diff
@@ -0,0 +1,19 @@
+import { Injectable, Scope } from '@nestjs/common';
+import { IntegrationRepository } from '@notifire/dal';
+import { RemoveIntegrationCommand } from './remove-integration.command';
+import { ApiException } from '../../../shared/exceptions/api.exception';
+
+@Injectable({
+  scope: Scope.REQUEST,
+})
+export class RemoveIntegration {
+  constructor(private integrationRepository: IntegrationRepository) {}
+
+  async execute(command: RemoveIntegrationCommand) {
+    try {
+      await this.integrationRepository.delete({ _id: command.integrationId });
+    } catch (e) {
+      throw new ApiException(e.message);
```

Comment:
Hey @scopsy  
Here I have a case where I want to catch an error that I throw on the DAL. However, it will much more convenient to catch a specific case (in my case  'integration that does not exist') and not all cases and rethrow them.

Reply:
Yeah, I think we discussed this earlier, we can just create DAL Exception and create an API middleware to handle this